dev/io.openliberty.cdi.4.0.internal.weld/src/io/openliberty/cdi40/internal/weld/LiteExtensionArchive.java

```diff
@@ -0,0 +1,181 @@
+    @Override
+    public ArchiveType getType() {
+        return ArchiveType.RUNTIME_EXTENSION;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ClassLoader getClassLoader() {
+        return LiteExtensionTranslator.class.getClassLoader();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Set<String> getClassNames() throws CDIException {
+        return Collections.singleton(LiteExtensionTranslator.class.getName());
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Set<String> getExtensionClasses() {
+        if (TraceComponent.isAnyTracingEnabled() && tc.isEventEnabled()) {
+            Tr.event(this, tc, "Returning extension translator");
+        }
+
+        return Collections.singleton(LiteExtensionTranslator.class.getName());
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Set<Supplier<Object>> getSPIExtensionSuppliers() {
+        return Collections.singleton(() -> new LiteExtensionTranslator(buildCompatibleExtensions, bceClassLoader));
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Resource getResource(String path) {
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean isModule() {
+        return false;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Application getApplication() {
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getClientModuleMainClass() throws CDIException {
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public List<String> getInjectionClassList() throws CDIException {
+        return Collections.emptyList();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public MetaData getMetaData() throws CDIException {
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public ResourceInjectionBag getAllBindings() throws CDIException {
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getClientAppCallbackHandlerName() throws CDIException {
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String getPath() throws CDIException {
+        return null;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Collection<CDIArchive> getModuleLibraryArchives() throws CDIException {
+        return Collections.emptySet();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Set<String> getAnnotatedClasses(Set<String> annotations) throws CDIException {
+        return Collections.emptySet();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Set<String> getExtraClasses() {
+        return Collections.emptySet();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public Set<String> getExtraBeanDefiningAnnotations() {
+        return Collections.emptySet();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean applicationBDAsVisible() {
+        return false; // TODO: might need to be true?
```

Comment:
is there a test to help decide this?

Reply:
I _suspect_ it will end up not needing to be `true` since I think extensions see everything anyway and this archive will always only contain a single extension.

Reply:
So any test which tried to enhance or inspect an application bean should cover this.

Reply:
Added an enhancement test and removed the TODO comment.